src/main/java/games/strategy/triplea/util/TuvUtils.java

```diff
@@ -81,11 +81,12 @@ private TuvUtils() {}
     }
 
     // Override with XML TUV or consumesUnit sum
+    final IntegerMap<UnitType> result = new IntegerMap<>(costs);
     for (final UnitType unitType : costs.keySet()) {
-      costs.put(unitType, getTotalTuv(unitType, costs, new HashSet<>()));
+      result.put(unitType, getTotalTuv(unitType, costs, new HashSet<>()));
```

Comment:
Added unit test. I think for loop vs stream is kind of a wash here.

Reply:
Note that the new test doesn't fail with the original implementation when using a `germanFortification`.  Not sure why, but some of the fortification unit types are not affected.  Final `costs` using the original code:

```
UnitType{name=Fortification} => 12
UnitType{name=americanFortification} => 12
UnitType{name=italianFortification} => 17
UnitType{name=turkishFortification} => 17
UnitType{name=brazilianFortification} => 17
UnitType{name=germanFortification} => 12
UnitType{name=japaneseFortification} => 17
UnitType{name=britishFortification} => 17
UnitType{name=russianFortification} => 17
UnitType{name=swedishFortification} => 17
UnitType{name=chineseFortification} => 17
UnitType{name=spanishFortification} => 17
```

Reply:
Well, its actually somewhat random failure as it depends what order the units go through the method. If entrenchment goes first then fort is 17 if fort goes first then its 12. But now after that change all should be 12.

Reply:
Ah, didn't realize that.  Would it then be safer to check all fortifications to avoid the non-determinism?

```java
@Test
public void testCostsForTuvWithConsumesUnitChain() {
  final PlayerID germans = GameDataTestUtil.germany(gameData);
  final IntegerMap<UnitType> result = TuvUtils.getCostsForTuv(germans, gameData);
  result.keySet().stream()
      .filter(unitType -> unitType.getName().endsWith("Fortification"))
      .forEach(unitType -> {
        assertEquals(12, result.getInt(unitType));
      });
}
```

Reply:
Well technically... They could all have been. It would potentially increase the odds...  I guess my point is that its kind of a hard case to write a real unit test for given the non-determinism. You would actually have to set the order of the unit types in the list to ensure entrenchment gets valued first.

Reply:
Point taken.  Then maybe my suggestion to add a test in the first place wasn't correct.  I'll go ahead and approve this and let you decide if keeping the new test is meaningful.

Reply:
The test is still a good unit test so we'll leave it.